frame/test_screencopy_bandwidth.py

```diff
@@ -26,29 +27,36 @@ def env(self) -> dict[str, str]:
 class TestScreencopyBandwidth:
     @pytest.mark.parametrize('server', all_servers())
     @pytest.mark.parametrize('app', [
-        (QTerminal('--execute', f'asciinema play {ASCIINEMA_CAST}'), 15),
+        (QTerminal('--execute', f'asciinema play {ASCIINEMA_CAST}', id='asciinema'), 15),
         (AppFixture("mir-kiosk-neverputt"), None),
-    ])
-    def test_active_app(self, server, app) -> None:
-        with DisplayServer(server) as s, app[0] as a:
-            with s.program(a) as p:
+    ], ids=lambda x: appids(x[0]))
+    def test_active_app(self, record_property, server, app) -> None:
+        with DisplayServer(server, add_extensions=ScreencopyTracker.required_extensions) as s, app[0] as a:
+            with ScreencopyTracker(s.display_name) as tracker, s.program(a) as p:
                 if app[1] is not None:
                     p.wait(app[1])
                 else:
                     time.sleep(long_wait_time)
+            for name, val in tracker.properties().items():
+                record_property(name, val)
 
     @pytest.mark.parametrize('server', all_servers())
-    def test_compositor_alone(self, server) -> None:
-        with DisplayServer(server) as s:
-            time.sleep(long_wait_time)
+    def test_compositor_alone(self, record_property, server) -> None:
+        with DisplayServer(server, add_extensions=ScreencopyTracker.required_extensions) as s:
+            with ScreencopyTracker(s.display_name) as tracker:
+                time.sleep(long_wait_time)
+            for name, val in tracker.properties().items():
+                record_property(name, val)
```

Comment:
Nah, that you shouldn't do. Sure, python doesn't have scopes, but it does have a garbage collector ;)

And the context manager is a… scope, of sorts.

Reply:
`__exit__()` gets called at the end of the `width` block, but `tracker` is still accessible and won't be arguable collected yet. `tracker.properties()` is written to be safe to call after `__exit__()`. Doing so is a good idea to make sure properties are retrieved after all frames have been processed.

Reply:
I'm under the impression it is specified behavior, but coming up with a source one way or the other is proving nontrivial, so I will do the more obvious thing and change it back.